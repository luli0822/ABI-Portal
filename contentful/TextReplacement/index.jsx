import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';

function TextReplacement({ text, replacements }) {
	const rootClass = 'textReplacement';
	const replacementsMap = {
		'â„¢': <sup className={`${rootClass}__sup`}>TM</sup>,
		'Â®': <sup className={`${rootClass}__sup`}>Â®</sup>,
		'ðŸ…ª': <sup className={`${rootClass}__sup`}>MC</sup>,
		'ðŸ…«': <sup className={`${rootClass}__sup`}>MD</sup>,
		...replacements,
	};

	const regexArray = Object.keys(replacementsMap);
	const regexFromArray = new RegExp(regexArray.join('|'), 'gi');
	const matches = text.match(regexFromArray);
	const newText = text.split(regexFromArray);

	const formattedText =
		newText &&
		newText.reduce((prev, curr, index) => {
			if (newText.length - 1 !== index) {
				const symbol = regexArray.find(key => key.toLowerCase() === matches[index].toLowerCase());

				if (symbol) {
					return [
						prev,
						curr,
						React.cloneElement(replacementsMap[symbol], {
							// eslint-disable-next-line react/no-array-index-key
							key: `text-replacement-${index}`,
						}),
					];
				}
			}

			return [...prev, curr];
		}, []);

	return formattedText || text;
}

TextReplacement.propTypes = {
	text: PropTypes.string.isRequired,
	replacements: PropTypes.shape({}),
};

TextReplacement.defaultProps = {
	replacements: {},
};

export default TextReplacement;
