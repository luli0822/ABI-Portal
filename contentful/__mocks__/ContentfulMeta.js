export const mockSiteMeta = {
	pageId: 'site-home-page',
	siteDomain: 'https://www.test-fr.ca/',
};

export const mockPageMeta = {
	pageId: 'about-us-page',
	siteDomain: 'https://www.test.ca',
};

export const mockNoFavicon = {
	sys: {
		id: 'siteId',
		contentType: {
			sys: {
				id: 'site',
				type: 'link',
			},
		},
	},
	fields: {
		title: {
			'en-CA': 'en title',
			'fr-CA': 'fr title',
		},
		indexPage: {
			'en-CA': {
				sys: {
					id: 'site-home-page',
				},
				fields: {
					slug: {
						'en-CA': 'index-slug',
					},
					title: {
						'en-CA': 'index title',
					},
				},
			},
		},
		pages: {
			'en-CA': [
				{
					sys: {
						id: 'about-us-page',
					},
				},
			],
		},
	},
};
