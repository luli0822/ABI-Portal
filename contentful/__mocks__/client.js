import fetchSiteData from './fetchSiteData.json';
// Refresh mock siteData will disclude about-us
module.exports = {
	getClient: () => {},
	fetchSiteData: () => fetchSiteData,
};
