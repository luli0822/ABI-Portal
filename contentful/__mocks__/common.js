// A method to build Link data structure
export const linkDataMock = (type, linkId = 'link-id-1') => ({
	sys: {
		id: linkId,
		contentType: {
			sys: {
				type: 'Link',
				linkType: 'ContentType',
				id: `${type}`,
			},
		},
	},
	fields: {
		title: {
			'en-CA': 'Link Text',
		},
		accessibleLabel: {
			'en-CA': 'Link accessible Text',
		},
		url: type === 'externalLink' ? { 'en-CA': 'https://loblawdigital.co' } : undefined,
		page:
			type === 'pageLink'
				? {
						'en-CA': {
							sys: {
								type: 'Link',
								linkType: 'Entry',
								id: 'site-home-page',
							},
						},
				  }
				: undefined,
	},
});

// Use this for pageLink structure mock
export const pageLinkMock = linkDataMock('pageLink');

// Use this externalLink structure mock
export const externalLinkMock = linkDataMock('externalLink');

// A method to build Media/image data structure
export const imageDataMock = type => ({
	sys: {
		type: 'Asset',
		id: type === 'image/png' ? 'asset-id-1' : 'asset-id-2',
	},
	fields: {
		title: {
			'en-CA': 'Image Title Text',
		},
		description: {
			'en-CA': 'Image description Text',
		},
		file: {
			'en-CA': {
				url: 'image-file-url.png',
				details: {
					size: 150795,
					image: {
						width: 300,
						height: 300,
					},
				},
				fileName: 'png-example.png',
				contentType: type,
			},
		},
	},
});
