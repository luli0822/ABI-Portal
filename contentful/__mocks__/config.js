module.exports = {
	space: 'spaceId',
	isPreview: true,
	accessToken: 'sometoken',
	host: 'preview.contentful.com',
	environment: 'master',
	siteSlug: 'site-slug',
	defaultLang: 'en-CA',
	fallbackLang: 'en-CA',
};
