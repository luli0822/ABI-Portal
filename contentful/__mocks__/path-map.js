module.exports = {
	pathMap: {
		'/': {
			page: '/',
			query: {
				lang: 'en-CA',
				pageId: 'site-home-page',
			},
		},
		'/home-pg-fr': {
			page: '/',
			query: {
				lang: 'fr-CA',
				pageId: 'site-home-page',
			},
		},
		'/about-pg-en': {
			page: '/',
			query: {
				lang: 'en-CA',
				pageId: 'about-us-page',
			},
		},
		'/about-pg-fr': {
			page: '/',
			query: {
				lang: 'fr-CA',
				pageId: 'about-us-page',
			},
		},
		// This path map for testing refreshed index page
		'/refreshed-page': {
			page: '/',
			query: {
				lang: 'en-CA',
				pageId: 'REFRESHED-site-home-page',
			},
		},
	},
};
