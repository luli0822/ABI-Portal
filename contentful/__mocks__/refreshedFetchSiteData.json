{
  "siteEntry": {
    "sys": {
      "space": {
        "sys": {
          "id": "spaceId"
        }
      },
      "id": "siteId",
      "contentType": {
        "sys": {
          "id": "site",
          "type": "link"
        }
      }
    },
    "fields": {
      "title": {
        "en-CA": "en title",
        "fr-CA": "fr title"
      },
      "slug": {
        "en-CA": "site-slug"
      },
      "indexPage": {
        "en-CA": {
          "sys": {
            "id": "REFRESHED-site-home-page"
          }
        }
      },
      "pages": {
        "en-CA": []
      },
      "keyValues": {
        "en-CA": [
          {
            "sys": {
              "type": "Link",
              "linkType": "Entry",
              "id": "lang-toggle-key-value"
            }
          },
          {
            "sys": {
              "type": "Link",
              "linkType": "Entry",
              "id": "rich-text-key-value"
            }
          }
        ]
      },
      "navigation": {
        "en-CA": [
          {
            "sys": {
              "type": "Link",
              "linkType": "Entry",
              "id": "some-external-link"
            }
          }
        ]
      }
    }
  },
  "linkedEntries": {
    "REFRESHED-site-home-page": {
      "sys": {
        "id": "REFRESHED-site-home-page",
        "space": {},
        "contentType": {
          "id": "page"
        }
      },
      "fields": {
        "title": {
          "en-CA": "title en",
          "fr-CA": "title fr"
        },
        "description": {
          "en-CA": "description en",
          "fr-CA": "description en"
        },
        "openGraphTitle": {
          "en-CA": "openGraphTitle en",
          "fr-CA": "openGraphTitle en"
        },
        "openGraphDescription": {
          "en-CA": "openGraphDescription en",
          "fr-CA": "openGraphDescription en"
        },
        "slug": {
          "en-CA": "home-pg-en",
          "fr-CA": "home-pg-fr"
        }
      }
    },
    "some-external-link": {
      "sys": {
        "id": "some-external-link",
        "space": {},
        "contentType": {
          "sys": {
            "id": "externalLink"
          }
        }
      },
      "fields": {
        "title": {
          "en-CA": "External link text",
          "fr-CA": "External link text fr"
        },
        "url": {
          "en-CA": "https://test.com",
          "fr-CA": "https://test.com/fr"
        }
      }
    },
    "lang-toggle-key-value": {
      "sys": {
        "id": "lang-toggle-key-value",
        "space": {},
        "contentType": {
          "id": "keyValue"
        }
      },
      "fields": {
        "slug": {
          "en-CA": "lang-toggle-slug"
        },
        "value": {
          "en-CA": "toggle English",
          "fr-CA": "toggle French"
        }
      }
    },
    "rich-text-key-value": {
      "sys": {
        "id": "rich-text-key-value",
        "space": {},
        "contentType": {
          "id": "keyRichText"
        }
      },
      "fields": {
        "slug": {
          "en-CA": "rich-text-key-slug"
        },
        "richText": {
          "en-CA": {
            "nodeType": "document",
            "data": {},
            "content": [
              {
                "nodeType": "paragraph",
                "data": {},
                "content": [
                  { "nodeType": "text", "data": {}, "value": "some text value" }
                ]
              }
            ]
          }
        }
      }
    },
    "some-complex-block-1": {
      "sys": {
        "id": "some-complex-block-1",
        "space": {},
        "contentType": {
          "id": "complexBlock"
        }
      },
      "fields": {
        "title": {
          "en-CA": "complex block title",
          "fr-CA": "complex block title - fr"
        },
        "body": {
          "en-CA": {
            "nodeType": "document",
            "data": {},
            "content": [
              {
                "nodeType": "paragraph",
                "data": {},
                "content": [
                  {
                    "nodeType": "text",
                    "data": {},
                    "value": "some paragraph in the complex block"
                  }
                ]
              }
            ]
          }
        }
      }
    },
    "some-complex-block-2": {
      "sys": {
        "id": "some-complex-block-2",
        "space": {},
        "contentType": {
          "id": "complexBlock"
        }
      },
      "fields": {
        "title": {
          "en-CA": "complex block 2 title",
          "fr-CA": "complex block 2 title - fr"
        },
        "body": {
          "en-CA": {
            "nodeType": "document",
            "data": {},
            "content": [
              {
                "nodeType": "paragraph",
                "data": {},
                "content": [
                  {
                    "nodeType": "text",
                    "data": {},
                    "value": "some paragraph in the complex block 2"
                  }
                ]
              }
            ]
          }
        }
      }
    },
    "some-block-group": {
      "sys": {
        "id": "some-block-group",
        "space": {},
        "contentType": {
          "id": "modularLayout"
        }
      },
      "fields": {
        "blocks": {
          "en-CA": [
            {
              "sys": {
                "type": "Link",
                "linkType": "Entry",
                "id": "some-complex-block-1"
              }
            },
            {
              "sys": {
                "type": "Link",
                "linkType": "Entry",
                "id": "some-complex-block-2"
              }
            }
          ]
        }
      }
    }
  },
  "linkedAssets": {
    "test-img.png": {
      "sys": {
        "id": "test-img.png"
      },
      "fields": {
        "title": {
          "en-CA": "some title",
          "fr-CA": "some title fr"
        },
        "description": {
          "en-CA": "some description",
          "fr-CA": "some description fr"
        },
        "file": {
          "en-CA": {
            "url": "/test-img.png",
            "fileName": "filename.png",
            "contentType": "image/png",
            "details": {
              "image": {
                "height": 20,
                "width": 20
              }
            }
          }
        }
      }
    },
    "test-img2.png": {
      "sys": {
        "id": "test-img2.png"
      },
      "fields": {
        "file": {
          "en-CA": {
            "url": "/test-img2.png",
            "fileName": "filename2.png",
            "contentType": "image/png",
            "details": {
              "image": {
                "height": 20,
                "width": 20
              }
            }
          }
        }
      }
    }
  }
}
