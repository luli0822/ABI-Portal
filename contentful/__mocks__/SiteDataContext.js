import pathMap from './path-map';
import { linkDataMock, imageDataMock } from './common';

export const mockLang = 'en-CA';

export const mockPathMap = pathMap.pathMap;

export const mockLinkedEntries = {
	'site-home-page': {
		sys: {
			id: 'site-home-page',
			space: {},
			contentType: {
				id: 'page',
			},
		},
		fields: {
			title: {
				'en-CA': 'title en',
				'fr-CA': 'title fr',
			},
			description: {
				'en-CA': 'description en',
				'fr-CA': 'description en',
			},
			openGraphTitle: {
				'en-CA': 'openGraphTitle en',
				'fr-CA': 'openGraphTitle en',
			},
			openGraphDescription: {
				'en-CA': 'openGraphDescription en',
				'fr-CA': 'openGraphDescription en',
			},
			slug: {
				'en-CA': 'home-pg-en',
				'fr-CA': 'home-pg-fr',
			},
		},
	},
	'about-us-page': {
		sys: {
			id: 'about-us-page',
			space: {},
			contentType: {
				id: 'page',
			},
		},
		fields: {
			title: {
				'en-CA': 'title en',
				'fr-CA': 'title fr',
			},
			description: {
				'en-CA': 'description en',
				'fr-CA': 'description en',
			},
			openGraphTitle: {
				'en-CA': 'openGraphTitle en',
				'fr-CA': 'openGraphTitle en',
			},
			openGraphDescription: {
				'en-CA': 'openGraphDescription en',
				'fr-CA': 'openGraphDescription en',
			},
			slug: {
				'en-CA': 'about-pg-en',
				'fr-CA': 'about-pg-fr',
			},
		},
	},
	'about-us-page-link': {
		sys: {
			id: 'about-us-page-link',
			space: {},
			contentType: {
				sys: {
					type: 'Link',
					id: 'pageLink',
				},
			},
		},
		fields: {
			title: {
				'en-CA': 'page link title',
				'fr-CA': 'page link title fr',
			},
			page: {
				'en-CA': {
					sys: {
						id: 'about-us-page',
					},
					fields: {
						slug: {
							'en-CA': 'some slug',
						},
					},
				},
			},
			hash: {
				'en-CA': 's',
			},
		},
	},
	'some-external-link': {
		sys: {
			id: 'some-external-link',
			space: {},
			contentType: {
				sys: {
					id: 'externalLink',
				},
			},
		},
		fields: {
			title: {
				'en-CA': 'External link text',
				'fr-CA': 'External link text fr',
			},
			accessibleLabel: {
				'en-CA': 'some accessibility text',
			},
			url: {
				'en-CA': 'https://test.com',
				'fr-CA': 'https://test.com/fr',
			},
		},
	},
	'lang-toggle-key-value': {
		sys: {
			id: 'lang-toggle-key-value',
			space: {},
			contentType: {
				id: 'keyValue',
			},
		},
		fields: {
			slug: {
				'en-CA': 'lang-toggle-slug',
			},
			value: {
				'en-CA': 'toggle English',
				'fr-CA': 'toggle French',
			},
		},
	},
	'rich-text-key-value': {
		sys: {
			id: 'rich-text-key-value',
			space: {},
			contentType: {
				id: 'keyRichText',
			},
		},
		fields: {
			slug: {
				'en-CA': 'rich-text-key-slug',
			},
			richText: {
				'en-CA': {
					nodeType: 'document',
					data: {},
					content: [
						{
							nodeType: 'paragraph',
							data: {},
							content: [{ nodeType: 'text', data: {}, value: 'some text value' }],
						},
					],
				},
			},
		},
	},
	'some-complex-block-1': {
		sys: {
			id: 'some-complex-block-1',
			space: {},
			contentType: {
				id: 'complexBlock',
			},
		},
		fields: {
			title: {
				'en-CA': 'complex block title',
				'fr-CA': 'complex block title - fr',
			},
			body: {
				'en-CA': {
					nodeType: 'document',
					data: {},
					content: [
						{
							nodeType: 'paragraph',
							data: {},
							content: [
								{ nodeType: 'text', data: {}, value: 'some paragraph in the complex block' },
							],
						},
					],
				},
			},
		},
	},
	'some-complex-block-2': {
		sys: {
			id: 'some-complex-block-2',
			space: {},
			contentType: {
				id: 'complexBlock',
			},
		},
		fields: {
			title: {
				'en-CA': 'complex block 2 title',
				'fr-CA': 'complex block 2 title - fr',
			},
			body: {
				'en-CA': {
					nodeType: 'document',
					data: {},
					content: [
						{
							nodeType: 'paragraph',
							data: {},
							content: [
								{ nodeType: 'text', data: {}, value: 'some paragraph in the complex block 2' },
							],
						},
					],
				},
			},
		},
	},
	'some-block-group': {
		sys: {
			id: 'some-block-group',
			space: {},
			contentType: {
				id: 'modularLayout',
			},
		},
		fields: {
			blocks: {
				'en-CA': [
					{
						sys: {
							type: 'Link',
							linkType: 'Entry',
							id: 'some-complex-block-1',
						},
					},
					{
						sys: {
							type: 'Link',
							linkType: 'Entry',
							id: 'some-complex-block-2',
						},
					},
				],
			},
		},
	},
};

export const mockSiteEntry = {
	sys: {
		space: {
			sys: {
				id: 'spaceId',
			},
		},
		id: 'siteId',
		contentType: {
			sys: {
				id: 'site',
				type: 'link',
			},
		},
	},
	fields: {
		title: {
			'en-CA': 'en title',
			'fr-CA': 'fr title',
		},
		slug: {
			'en-CA': 'site-slug',
		},
		siteLogo: {
			'en-CA': {
				fields: {
					file: {
						'en-CA': {
							url: '/test-img.png',
							fileName: 'filename.png',
							contentType: 'image/png',
							details: {
								image: {
									height: 10,
									width: 10,
								},
							},
						},
					},
					title: {
						'en-CA': 'image title',
					},
				},
			},
		},
		favicon: {
			'en-CA': {
				sys: {
					id: 'favicon.png',
				},
				fields: {
					file: {
						'en-CA': {
							url: '/favicon.png',
							fileName: 'favicon.png',
							contentType: 'image/png',
							details: {
								image: {
									height: 10,
									width: 10,
								},
							},
						},
					},
				},
			},
		},
		indexPage: {
			'en-CA': {
				sys: {
					id: 'site-home-page',
				},
				fields: {
					slug: {
						'en-CA': 'index-slug',
					},
					title: {
						'en-CA': 'index title',
					},
				},
			},
		},
		pages: {
			'en-CA': [
				{
					sys: {
						id: 'about-us-page',
					},
				},
			],
		},
		keyValues: {
			'en-CA': [
				{
					sys: {
						type: 'Link',
						linkType: 'Entry',
						id: 'lang-toggle-key-value',
					},
				},
				{
					sys: {
						type: 'Link',
						linkType: 'Entry',
						id: 'rich-text-key-value',
					},
				},
			],
		},
		navigation: {
			'en-CA': [
				{
					sys: {
						type: 'Link',
						linkType: 'Entry',
						id: 'about-us-page-link',
					},
				},
				{
					sys: {
						type: 'Link',
						linkType: 'Entry',
						id: 'some-external-link',
					},
				},
			],
		},
	},
};

export const mockLinkedAssets = {
	'test-img.png': {
		sys: {
			id: 'test-img.png',
		},
		fields: {
			title: {
				'en-CA': 'some title',
				'fr-CA': 'some title fr',
			},
			description: {
				'en-CA': 'asset description',
				'fr-CA': 'asset description fr',
			},
			file: {
				'en-CA': {
					url: '/test-img.png',
					fileName: 'filename.png',
					contentType: 'image/png',
					details: {
						image: {
							height: 20,
							width: 20,
						},
					},
				},
			},
		},
	},
	'test-img2.png': {
		sys: {
			id: 'test-img2.png',
		},
		fields: {
			// no description - used to test image components
			file: {
				'en-CA': {
					url: '/test-img2.png',
					fileName: 'filename2.png',
					contentType: 'image/png',
					details: {
						image: {
							height: 20,
							width: 20,
						},
					},
				},
			},
		},
	},
	'favicon.png': {
		sys: {
			id: 'favicon.png',
		},
		fields: {
			file: {
				'en-CA': {
					url: '/favicon.png',
					fileName: 'favicon.png',
					contentType: 'image/png',
					details: {
						image: {
							height: 20,
							width: 20,
						},
					},
				},
			},
		},
	},
};

export const mockPreviewAssets = {
	'test-img.png': { url: 'data:image/png;base64,someUriData' },
	'test-img2.png': { url: 'data:image/png;base64,someUriData2' },
};

export const mockKeyValues = {
	'lang-toggle-slug': {
		sys: {
			id: 'lang-toggle-key-value',
			space: {},
			contentType: {
				id: 'keyValue',
			},
		},
		fields: {
			slug: {
				'en-CA': 'lang-toggle-slug',
			},
			value: {
				'en-CA': 'toggle English',
				'fr-CA': 'toggle French',
			},
		},
	},
	'rich-text-key-slug': {
		sys: {
			id: 'rich-text-key-value',
			space: {},
			contentType: {
				id: 'keyRichText',
			},
		},
		fields: {
			slug: {
				'en-CA': 'rich-text-key-slug',
			},
			richText: {
				'en-CA': {
					nodeType: 'document',
					data: {},
					content: [
						{
							nodeType: 'paragraph',
							data: {},
							content: [{ nodeType: 'text', data: {}, value: 'some text value' }],
						},
					],
				},
			},
		},
	},
};

export const mockLogo = {
	primary: imageDataMock('image/png'),
	primaryAccessibleText: 'Accessible description of logo',
	alt: imageDataMock('image/svg+xml'),
	altAccessibleText: 'Accessible description of logo',
};

export const mockHeaderNav = [
	{
		type: 'link',
		link: linkDataMock('pageLink', 'nav-id-1'),
		text: 'Nav 1',
		accessibleText: 'Nav 1 accessible text',
	},
	{
		type: 'link',
		link: linkDataMock('pageLink', 'nav-id-2'),
		text: 'Nav 2',
		accessibleText: 'Nav 2 accessible text',
	},
];

export const mockFooterNav = [
	{
		type: 'link',
		link: linkDataMock('pageLink', 'footer-nav-id-1'),
		text: 'Nav 1',
		accessibleText: 'Nav 1 accessible text',
	},
];

export const mockLangSwitchLink = {
	href: '/some-href',
	asPath: '/some-path',
	text: 'change lang',
	accessibleText: 'Switch lang to EN',
};

export const mockSkipNavText = 'Skip to main content';

function isLink(field) {
	return field && field.sys && field.sys.type === 'Link';
}

export const mockFieldOrFallback = jest.fn((field, fieldLang = 'en-CA') => {
	if (isLink(field)) {
		if (field.sys.linkType === 'Entry') return mockLinkedEntries[field.sys.id];
		if (field.sys.linkType === 'Asset') return mockLinkedAssets[field.sys.id];
	}

	const fieldValue = field ? field[fieldLang] || field['en-CA'] : undefined;

	if (isLink(fieldValue)) {
		if (fieldValue.sys.linkType === 'Entry') return mockLinkedEntries[fieldValue.sys.id];
		if (fieldValue.sys.linkType === 'Asset') return mockLinkedAssets[fieldValue.sys.id];
	}

	return fieldValue;
});

export const mockRichText = {
	'without-markings': {
		data: {},
		content: [
			{
				data: {},
				content: [
					{
						data: {},
						marks: [],
						value: 'Text without markings italicized',
						nodeType: 'text',
					},
				],
				nodeType: 'paragraph',
			},
		],
		nodeType: 'document',
	},
	'with-markings': {
		data: {},
		content: [
			{
				data: {},
				content: [
					{
						data: {},
						marks: [{ type: 'bold' }],
						value: 'Bold Text',
						nodeType: 'text',
					},
					{ data: {}, marks: [], value: ' for ', nodeType: 'text' },
					{
						data: {},
						marks: [{ type: 'italic' }],
						value: 'Italic Text',
						nodeType: 'text',
					},
				],
				nodeType: 'paragraph',
			},
			{
				nodeType: 'unordered-list',
				content: [
					{
						nodeType: 'list-item',
						data: {},
						content: [
							{
								nodeType: 'paragraph',
								content: [
									{
										nodeType: 'text',
										value: 'point 1',
										marks: [],
										data: {},
									},
								],
								data: {},
							},
						],
					},
					{
						nodeType: 'list-item',
						content: [
							{
								nodeType: 'paragraph',
								content: [
									{
										nodeType: 'text',
										value: 'point 2',
										marks: [],
										data: {},
									},
								],
								data: {},
							},
						],
						data: {},
					},
				],
				data: {},
			},
		],
		nodeType: 'document',
	},
	'with-link': {
		data: {},
		content: [
			{
				data: {},
				content: [
					{
						data: {},
						marks: [{ type: 'bold' }],
						value: 'We bring marketing sites to life',
						nodeType: 'text',
					},
					{ data: {}, marks: [], value: ' for ', nodeType: 'text' },
					{
						data: {},
						marks: [{ type: 'italic' }],
						value: 'brands',
						nodeType: 'text',
					},
					{
						data: {},
						marks: [],
						value: ' across the organization. Learn more ',
						nodeType: 'text',
					},
					{
						data: {
							target: {
								'en-CA': {
									sys: {
										id: 'about-us-page-link',
										contentType: {
											sys: {
												id: 'externalLink',
											},
										},
									},
									fields: {
										title: {
											'en-CA': 'mock link title italicized',
										},
										url: {
											'en-CA': 'www.site.com',
										},
										accessibleLabel: {
											'en-CA': 'accessible text ',
										},
									},
								},
							},
						},
						content: [],
						nodeType: 'embedded-entry-inline',
					},
					{ data: {}, marks: [], value: '', nodeType: 'text' },
				],
				nodeType: 'paragraph',
			},
			{
				data: {},
				content: [{ data: {}, marks: [], value: '', nodeType: 'text' }],
				nodeType: 'paragraph',
			},
			{
				data: {},
				content: [{ data: {}, marks: [], value: '', nodeType: 'text' }],
				nodeType: 'paragraph',
			},
		],
		nodeType: 'document',
	},
};
