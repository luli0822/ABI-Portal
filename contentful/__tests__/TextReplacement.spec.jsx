import React from 'react';
import { mount } from 'enzyme';
import TextReplacement from '../TextReplacement';

import { mockLang, mockFieldOrFallback } from '../__mocks__/SiteDataContext';

// Mock React useContext
jest.mock('react', () => {
	const ActualReact = require.requireActual('react');
	return {
		...ActualReact,
		useContext: () => ({
			fieldOrFallback: mockFieldOrFallback,
			lang: mockLang,
		}),
	};
});

const mockProps = {
	text: 'This is just regularText',
};

const mockPropsWithReplacements = {
	text: 'This is a tradmark ™ and this worked is italicized',
	replacements: {
		'™': <sup className="sup">™</sup>,
		italicized: <span className="i">italicized</span>,
	},
};

describe('<ContentfulImage />', () => {
	it('Renders just text with no markup', () => {
		// eslint-disable-next-line react/jsx-props-no-spreading
		const component = mount(<TextReplacement {...mockProps} />);

		expect(component.debug()).toMatchSnapshot();
	});
	it('Renders text with correct markup added', () => {
		// eslint-disable-next-line react/jsx-props-no-spreading
		const component = mount(<TextReplacement {...mockPropsWithReplacements} />);

		expect(component.debug()).toMatchSnapshot();
	});
});
