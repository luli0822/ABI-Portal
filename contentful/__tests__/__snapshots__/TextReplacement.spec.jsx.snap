// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentfulImage /> Renders just text with no markup 1`] = `
"<TextReplacement text=\\"This is just regularText\\" replacements={{...}}>
  This is just regularText
</TextReplacement>"
`;

exports[`<ContentfulImage /> Renders text with correct markup added 1`] = `
"<TextReplacement text=\\"This is a tradmark ™ and this worked is italicized\\" replacements={{...}}>
  This is a tradmark 
  <sup className=\\"sup\\">
    ™
  </sup>
   and this worked is 
  <span className=\\"i\\">
    italicized
  </span>
</TextReplacement>"
`;
